src/facilities-context/FacilitiesContext.tsx

```diff
@@ -69,24 +66,44 @@ export const FacilitiesProvider: React.FC<{ children: React.ReactNode }> = ({
     selectedFacilityId: null,
     rtData: {},
   });
-  const [scenario] = useScenario();
-  const scenarioId = scenario?.data?.id;
+  const [scenarioState] = useScenario();
+  const shouldUseReferenceFacilities = useReferenceFacilitiesEligible();
+  const scenario = scenarioState.data;
+  const scenarioId = scenario?.id;
+  const facilityToReference =
+    scenario && shouldUseReferenceFacilities
+      ? scenario[referenceFacilitiesProp]
+      : undefined;
+
   const actions = {
-    createOrUpdateFacility: facilitiesActions.createOrUpdateFacility(dispatch),
+    createOrUpdateFacility: facilitiesActions.createOrUpdateFacility(
+      dispatch,
+      scenario,
+      state.referenceFacilities,
+    ),
     fetchFacilityRtData: facilitiesActions.fetchFacilityRtData(dispatch),
     removeFacility: facilitiesActions.removeFacility(dispatch),
-    duplicateFacility: facilitiesActions.duplicateFacility(dispatch),
+    duplicateFacility: facilitiesActions.duplicateFacility(
+      dispatch,
+      scenario,
+      state.referenceFacilities,
+    ),
     deselectFacility: facilitiesActions.deselectFacility(dispatch),
     selectFacility: facilitiesActions.selectFacility(dispatch),
   };
 
   useEffect(() => {
-    if (scenarioId) {
-      facilitiesActions.fetchFacilities(scenarioId, dispatch);
+    if (scenario) {
       // clean up any existing reference facility data
       facilitiesActions.clearReferenceFacilities(dispatch);
+
+      facilitiesActions.fetchFacilities(
+        shouldUseReferenceFacilities,
+        dispatch,
+        scenario,
+      );
     }
-  }, [scenarioId]);
+  }, [facilityToReference, scenario, shouldUseReferenceFacilities]);
```

Comment:
Would this mean whenever a user syncs a new facility to reference data, we would re-fetch all facilities and recreate all the composite facilities? I wonder if this will play well if we end up adding actions specific to updating/syncing a single facility from a modal, since that would also update the `facilityToReference` on the `scenario`.

Reply:
interesting ... that wasn't the intention. ideally we would not re-fetch the data, we would just rebuild the Facility object for the newly synced facility. i will look into this some more

Reply:
I was thinking about this and trying to think of a way to not depend on the scenario for syncing the facilities to reference facilities. What about having another `useEffect` that depends on scenario and calls `fetchReferenceFacilities` and stores those in the state as `{ facilityId: ReferenceFacility }` using the scenario, so we already have the `facilityId` it maps to in the store and can avoid passing `scenario` to `getReferenceFacility` later on. 

Then we could have another `useEffect` that depends on `facilities` and `referenceFacilities` and calls the `mergeFacilityObjects` and `getReferenceFacility` for each of those. I haven't tested this out, just a thought I had this morning, let me know if it doesn't make sense!

Reply:
interesting ... at some point I tried to do something similar to this but I couldn't get it to work. I'm going to pull at this thread again though because I think there's some potential

Reply:
Okay cool, let me know if you want to talk through it at all